resteasy-core/src/main/java/org/jboss/resteasy/core/ContextParameterInjector.java

```diff
@@ -195,4 +209,50 @@ public ClassLoader run() {
          return Proxy.newProxyInstance(clazzLoader, intfs, new GenericDelegatingProxy());
       }
    }
+
+   public boolean outputStreamWasWrittenTo()
+   {
+      return outputStreamWasWritten;
+   }
+
+   public class ContextOutputStream extends ServletOutputStream
+   {
+      private ServletOutputStream delegate;
+
+      public ContextOutputStream(final ServletOutputStream delegate)
+      {
+         this.delegate = delegate;
+      }
+
+      @Override
+      public void write(int b) throws IOException
+      {
+         delegate.write(b);
+         outputStreamWasWritten = true;
+      }
```

Comment:
Is there a reason why you're not overriding `public void write(byte b[], int off, int len)` method as well?

Reply:
Well, write(byte b[]) and write(byte b[], int off, int len) both end up calling write():
```
    public void write(byte b[]) throws IOException {
        write(b, 0, b.length);
    }
   ...
    public void write(byte b[], int off, int len) throws IOException {
        Objects.checkFromIndexSize(off, len, b.length);
        // len == 0 condition implicitly handled by loop bounds
        for (int i = 0 ; i < len ; i++) {
            write(b[off + i]);
        }
    }
```

Reply:
In the default implementation they do, but we can't know for sure that all delegates will do that. It could be buffered as well meaning `outputStreamWasWritten` won't be changed to `true` until a flush happens.

Reply:
Ah, so someone could extend ServletOutputStream and overwrite the other write()'s. Ok, thanks.

Reply:
Potentially yes. IMO any delegating class should override and delegate all methods. Assumptions can't be made about the delegate.

Reply:
Mmmm, yeah, ok. By "all methods" ... if someone overrides ServletOutputStream, it's really unknowable what other methods they could have added ... . I'm not sure what to do about that ... .

Reply:
Well, not exactly unknowable ...

Reply:
As long as we override each method we don't have to care what they do :) We just simply delegate.

Reply:
But suppose they add a new method that doesn't override some method in ServletOutputStream or OutputStream? I could find it by reflection, but then what? Maybe ContextOutputStream should itself be a proxy, which could catch all methods.

This is a job for Python. ;-)

Reply:
Well we can't really do anything about that. We only care about the base class we're wrapping :) And IMO no, we don't want a proxy for performance reasons. If a container adds it's own methods that aren't accessible from the base type that's on them :D

Reply:
Yeah, a proxy in a proxy. 

Ok, ServletOutputStream has 17 methods. :)

Here I go.